File: backend/api-core/internal/domain/vm.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type VMStatus string

const (
	StatusEncendida VMStatus = "encendida"
	StatusApagada   VMStatus = "apagada"
)

var ValidOS = [5]string{
	"Ubuntu 22.04",
	"Windows Server 2022",
	"CentOS 8",
	"Debian 12",
	"Rocky Linux 9",
}

type VM struct {
	ID        uuid.UUID
	Name      string
	Cores     int
	RAMGB     int
	DiskGB    int
	OS        string
	Status    VMStatus
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: frontend/src/types/vm.ts
export type VMStatus = 'encendida' | 'apagada';

export interface VM {
  id: string;
  name: string;
  cores: number;
  ram_gb: number;
  disk_gb: number;
  os: string;
  status: VMStatus;
  created_at: string;
  updated_at: string;
}

export interface VMFormData {
  name: string;
  cores: number;
  ram_gb: number;
  disk_gb: number;
  os: string;
  status: VMStatus;
}

export const OS_OPTIONS = [
  'Ubuntu 22.04',
  'Windows Server 2022',
  'CentOS 8',
  'Debian 12',
  'Rocky Linux 9',
] as const;
